packages/cli/src/modules/insights/__tests__/insights.service.test.ts

```diff
@@ -664,3 +671,105 @@ describe('getMaxAgeInDaysAndGranularity', () => {
 		});
 	});
 });
+
+describe('shutdown', () => {
+	let insightsService: InsightsService;
+
+	const mockCollectionService = {
+		shutdown: jest.fn().mockResolvedValue(undefined),
+		stopFlushingTimer: jest.fn(),
+	} as unknown as InsightsCollectionService;
+
+	const mockCompactionService = {
+		stopCompactionTimer: jest.fn(),
+	} as unknown as InsightsCompactionService;
+
+	const mockPruningService = {
+		stopPruningTimer: jest.fn(),
+	} as unknown as InsightsPruningService;
+
+	beforeAll(() => {
+		insightsService = new InsightsService(
+			mock<InsightsByPeriodRepository>(),
+			mockCompactionService,
+			mockCollectionService,
+			mockPruningService,
+			mock<LicenseState>(),
+			mockLogger(),
+		);
+	});
+
+	test('shutdown stops timers and shuts down services', async () => {
+		// ACT
+		await insightsService.shutdown();
+
+		// ASSERT
+		expect(mockCollectionService.shutdown).toHaveBeenCalled();
+		expect(mockCompactionService.stopCompactionTimer).toHaveBeenCalled();
+		expect(mockPruningService.stopPruningTimer).toHaveBeenCalled();
+	});
+});
+
+describe('backgroundProcess', () => {
```

Comment:
I wonder, is it really a "background process"? As it's not really a child process via `fork` or `spawn`.

Reply:
Yes you're right, it's misleading. I renamed for timers for now (tests and service methods), as that's all the functions do for now (start timeout / intervals)